phone/src/apps/messages/components/ui/MessageEmbed.tsx

```diff
@@ -14,53 +14,73 @@ interface MessageEmbedProps {
   type: string;
   embed: any;
   isMine: boolean;
+  message: string;
 }
 
 type MessageEmbedType = {
   [key: string]: JSX.Element;
 };
 
-const MessageEmbed: React.FC<MessageEmbedProps> = ({ type, embed, isMine }) => {
+const MessageEmbed: React.FC<MessageEmbedProps> = ({ type, embed, isMine, message }) => {
   const embedType: MessageEmbedType = {
-    contact: <ContactEmbed embed={embed} isMine={isMine} />,
-    location: <LocationEmbed embed={embed} isMine={isMine} />,
+    contact: <ContactEmbed embed={embed} isMine={isMine} message={message} />,
+    location: <LocationEmbed embed={embed} isMine={isMine} message={message} />,
   };
 
   return <>{embedType[type]}</>;
 };
 
-const ContactEmbed = ({ isMine, embed }: { isMine: boolean; embed: Contact }) => {
+const ContactEmbed = ({
+  isMine,
+  embed,
+  message,
+}: {
+  isMine: boolean;
+  embed: Contact;
+  message: string;
+}) => {
   const [t] = useTranslation();
   const history = useHistory();
   const { pathname } = useLocation();
   const { getContactByNumber } = useContactActions();
 
   const showAddButton = !isMine && !getContactByNumber(embed?.number);
+  const showMessage = message !== t('MESSAGES.CONTACT_SHARED');
 
   const handleAddContact = () => {
     const referal = encodeURIComponent(pathname);
     history.push(`/contacts/-1?addNumber=${embed.number}&name=${embed.display}&referal=${referal}`);
   };
 
   return (
-    <StyledMessage>
-      <Box>
-        <Avatar src={embed?.avatar} />
-        <Typography>{embed?.display}</Typography>
-        <Typography>{embed?.number}</Typography>
-      </Box>
-      {showAddButton && (
+    <StyledMessage flexDirection="column">
+      {showMessage && <Box sx={{ width: '100%' }}>{message}</Box>}
```

Comment:
Does this need a whole box with 100% width? 
![image](https://user-images.githubusercontent.com/27494805/167210202-c40eb564-92c8-4d56-9408-811a061029fb.png)
You have a lot of empty space here

Reply:
I tried to follow as most as I could the original design, the issue I had was with the `ADD` button position while having a message above.

Reply:
I'm not too sure about having messages show up in these embeds, it looks very out of place imo

Reply:
Tbh I only added it in the ContactEmbed because I could, I don't really see the same use as for the LocationEmbed. I can remove if you prefer.

Reply:
I think that would be for the better, in the location embeds I can see the case for it yeah

Reply:
Ok I'll remove it then

Reply:
Changed in last commit